flink-runtime/src/main/java/org/apache/flink/runtime/taskexecutor/TaskManagerServices.java

```diff
@@ -197,7 +201,13 @@ public static TaskManagerServices fromConfiguration(
 		final JobManagerTable jobManagerTable = new JobManagerTable();
 
 		final JobLeaderService jobLeaderService = new JobLeaderService(taskManagerLocation);
-		final TaskExecutorLocalStateStoresManager taskStateManager = new TaskExecutorLocalStateStoresManager();
+
+		final File taskExecutorLocalStateRootDir =
+			new File(taskManagerServicesConfiguration.getLocalStateRootDir(), LOCAL_STATE_SUB_DIRECTORY_ROOT);
```

Comment:
Are we giving the `TaskExecutorLocalStateStoresManager` a unique directory? What happens if multiple TMs run on the same machine? Will they use different directories?

Reply:
No, they can share the same directories, because inside things are further grouped into per-allocation directories.